app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,102 @@
+class TweetsController < ApplicationController
+
+  configure do
+    enable :sessions
+    set :session_secret, "secret"
+  end
+
+  get '/tweets' do
+    if logged_in?
+      @users = current_user
+      @tweets = Tweet.all
+      erb :'/tweets/tweets'
+    else
+      redirect to "/login"
+    end
+
+  end
+
+
+  get '/tweets/new' do
+    if logged_in?
+      @user = current_user
+      erb :'tweets/create_tweet'
+    else
+      redirect "/login"
+    end
+  end
+
+  post '/tweets' do
+    @tweet = Tweet.new(params)
+    @user = current_user
+    #check if tweet is empty.
+    # This is where we set the name for song/ it want us to pass in an hash.
+    if logged_in? && !@tweet.content.blank? && @tweet.save
+      @user.tweets << @tweet
+      redirect to "/tweets/#{@tweet.id}"  # target Id of specific tweets.
+    else
+      redirect "/tweets/new"  #this is a route
+    end
+  end
+
+
+  get "/tweets/:id" do
+    if logged_in?
+      @tweet = Tweet.find_by_id(params[:id])
+      erb :'/tweets/show_tweet'
+    else
+      redirect to "/login"
+    end
+
+  end
+
+
+  get '/tweets/:id/edit' do
+
+     if logged_in?
+    @tweet = Tweet.find_by_id(params[:id]) # slug helps to find by name instaed of ID
+    erb :'tweets/edit_tweet'
+   else
+
+    redirect to "/login"
+
+    end
+
+  end
+
+  # <this connect here
+
+  patch "/tweets/:id" do
+    @tweet = Tweet.find(params[:id])
+
+        if logged_in? && !params[:content].blank?
+           @tweet.update(content: params[:content])  # shovel in Title into figure.titles to be used in the views folder
+        # :Note I can explicit with each.. but figures contains everthing already
+        @tweet.save
+
+        redirect to "/tweets/#{@tweet.id}"
+      else
+        redirect to "/tweets/#{@tweet.id}/edit"
+      end
+
+    end
+
+
+  delete "/tweets/:id/delete" do
+    @tweet = Tweet.find_by_id(params[:id])
+
+    if logged_in? && @tweet.user == current_user
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.